hack/e2e-test.sh

```diff
@@ -28,14 +28,15 @@ export E2E_TEST_IMAGE=gcr.io/k8s-staging-perf-tests/sleep:v0.1.0
 source ${SOURCE_DIR}/e2e-common.sh
 
 function cleanup {
+    EX_CODE=$?
     if [ $CREATE_KIND_CLUSTER == 'true' ]
     then
         if [ ! -d "$ARTIFACTS" ]; then
             mkdir -p "$ARTIFACTS"
         fi
 	cluster_cleanup $KIND_CLUSTER_NAME
     fi
-    exit 0
+    exit $EX_CODE
```

Comment:
IIUC this will cover if there is an error from `cluster_cleanup`. I think it would be preferable to fail the command in that case as well. Could be useful to check if tests in k8s fail or pass in case of cleanup failure.

Reply:
Let's wait and see what the intent of #1730 was, I think it was to ignore potential errors during cleanup. 
@alculquicondor ?

Reply:
The intent was to allow `@$(call clean-manifests)` in the Makefile to run, whether on exit or success.

Reply:
Maybe we can use `.ONESHELL` and `$?` like here? https://stackoverflow.com/a/44324918

Reply:
for e2e-s we should do the `kustomize` clenup in the hack scripts since there we are making it dirty also

https://github.com/kubernetes-sigs/kueue/blob/74d17e38e76c2d8b47dbea74364aecc5529f3a78/hack/e2e-test.sh#L60

maybe we can do it at the end of `kueue_deploy` before running the tests

Reply:
I'm trying to avoid duplication and having to pass another environment variable (it's not `$IMAGE_TAG`).

Reply:
We get the  current image before running, and restore to that